pkg/dio/multi.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dio adds dynamic behaviour to the standard io package mutliX types
+package dio
+
+import (
+	"io"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+var verbose = true
+
+// DynamicMultiWriter adds dynamic add/remove to the base multiwriter behaviour
+type DynamicMultiWriter interface {
+	io.Writer
```

Comment:
I'll give it a go. I had issues when I made this a WriteCloser at one point but I don't recall why - I think it's because not all writers that get added are WriteClosers and I was unsure about mixing the semantics.